src/elements.jl

```diff
@@ -145,3 +145,20 @@ function local_potential_fourier(el::ElementCohenBergstresser, q::T) where {T <:
     qsq_pi = Int(round(q^2 / (2π / el.lattice_constant)^2, digits=2))
     T(get(el.V_sym, qsq_pi, 0.0))
 end
+
+"""
+Element for an ion with charge Z that generates a custom local potential.
+There is no check of the relation between pot_real and pot_fourier, the user
+has to ensure that point.
+"""
+struct ElementCustomIonPotential <: Element
+    Z::Int                  # Nuclear charge
+    pot_real::Function      # Real potential
+    pot_fourier::Function   # Fourier potential
+end
+charge_ionic(el::ElementCustomIonPotential) = el.Z
```

Comment:
For this model, the nuclear charge (which is really the ionic charge, corresponding to the number of valence electrons only) does not really make sense, so I would rather leave it unimplemented and not have a Z field

Reply:
I agree but I think we still need to have a method `charge_ionic(el::ElementCustomIonPotential)` to avoid errors when using this element ?

Reply:
It should only get called if n_electrons is not specified.

Reply:
Even with `n_electrons = 1` in the example, I get the error `Implement charge_ionic` if I don't keep that line.

Reply:
Can you post a backtrace?

Reply:
```
ERROR: LoadError: Implement charge_ionic
Stacktrace:
 [1] charge_ionic(::ElementCustomIonPotential) at /home/kemling/Documents/Nextcloud/CERMICS/codes/DFTK/DFTK.jl/src/elements.jl:16
 [2] n_elec_valence(::ElementCustomIonPotential) at /home/kemling/Documents/Nextcloud/CERMICS/codes/DFTK/DFTK.jl/src/elements.jl:19
 [3] macro expansion at /home/kemling/Documents/Nextcloud/CERMICS/codes/DFTK/DFTK.jl/src/guess_density.jl:20 [inlined]
 [4] (::DFTK.var"#476#477"{Float64,PlaneWaveBasis{Float64},Array{Pair,1}})() at /home/kemling/.julia/packages/TimerOutputs/dVnaw/src/TimerOutput.jl:237
 [5] timeit(::DFTK.var"#476#477"{Float64,PlaneWaveBasis{Float64},Array{Pair,1}}, ::TimerOutputs.TimerOutput, ::String) at /home/kemling/.julia/packages/TimerOutputs/dVnaw/src/TimerOutput.jl:285
 [6] guess_density at /home/kemling/.julia/packages/TimerOutputs/dVnaw/src/TimerOutput.jl:236 [inlined]
 [7] guess_density at /home/kemling/Documents/Nextcloud/CERMICS/codes/DFTK/DFTK.jl/src/guess_density.jl:12 [inlined]
 [8] top-level scope at /home/kemling/Documents/Nextcloud/CERMICS/codes/DFTK/DFTK.jl/examples/gross_pitaevskii_custom_potential.jl:45
 [9] top-level scope at REPL[4]:1
 [10] run_backend(::REPL.REPLBackend) at /home/kemling/.julia/packages/Revise/kqqw8/src/Revise.jl:1163
 [11] top-level scope at none:0
in expression starting at /home/kemling/Documents/Nextcloud/CERMICS/codes/DFTK/DFTK.jl/examples/gross_pitaevskii_custom_potential.jl:45
```
(Indeed, the timer pollutes the stacktrace :stuck_out_tongue: )

Reply:
Ah yes, the guess density. We should probably switch to direction minimization here anyway... You can bypass this by giving an explicit density.